solutions/cses-1084.mdx

```diff
@@ -54,3 +73,66 @@ int main(){
 	return 0;
 }
 ```
+
+</CPPSection>
+
+<JavaSection>
+
+<Warning>
+Java will TLE on test case #17.
+</Warning>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Apartments {
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();
+		int m = io.nextInt();
+		int k = io.nextInt();
```

Comment:
better names here too

Reply:
these are from the problem statement though...

Reply:
don't think anyone wants to constantly flip from the problem statement to the solution code over & over again

Reply:
If they're trying to solve the problem, won't they already know what all of the variables mean?

Reply:
think most people just look at the problem variable names (like "the number of apartments") and go right to thinking + coding

Reply:
^

Reply:
woops that was meant to point towards @nathangong

Reply:
lmao

Reply:
> think most people just look at the problem variable names (like "the number of apartments") and go right to thinking + coding

Well the input format is given in terms of n, m, and k so I think it's safe to assume they know what is what given that they have to deal with them to read in the input. Plus, you generally use the variable names n, m, and k when you're trying to determine what the time complexity of your approach is.

Reply:
still think it'd be better to be explicit about the variable names
`peopleNum`, `aptNum`, and `maxDiff` or smth
also what's `a` and `b`

Reply:
Alright I can compromise by adding comments for each variable explaining what they do. `a` stores the desired appartment size of each applicant and `b` stores the size of each apartment as explained in the problem.

Reply:
yeah cool i'll accept that